Creating vms in modules leads to terraform-provider-proxmox crash

Hi, thanks to everyone for your work on this provider ! 
Meanwhile, I've faced an issue when I tried to organize my code.

**Bug Description**
I wrote my whole terraform process in a main.tf file, to begin with, and everything works perfectly. So I tried to organize myself and split what I did to a few modules. But when I do so, a panic error shows : 

```
│ Error: Plugin did not respond
│
│ The plugin encountered an error, and failed to respond to the plugin6.(*GRPCProvider).ApplyResourceChange call. The
│ plugin logs may contain more details.

Stack trace from the terraform-provider-proxmox_v0.50.0.exe plugin:

panic: interface conversion: interface {} is nil, not string

goroutine 19 [running]:
github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource/vm.vmGetCloudInitConfig(0xc00017ce00?)
        github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource/vm/vm.go:2665 +0x103d
github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource/vm.vmCreateCustom({0x1510cb0, 0xc0006b1420}, 0xc00017ce00, {0x12c86e0, 0xc0000f5350})
        github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource/vm/vm.go:2241 +0x12c8
github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource/vm.vmCreate({0x1510cb0, 0xc0006b1420}, 0xc00017ce00, {0x12c86e0, 0xc0000f5350})
        github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource/vm/vm.go:1400 +0x94
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc0002cfdc0, {0x1510c08, 0xc000580d80}, 0xc00017ce00, {0x12c86e0, 0xc0000f5350})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.33.0/helper/schema/resource.go:778 +0x119
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0002cfdc0, {0x1510c08, 0xc000580d80}, 0xc000455d40, 0xc00017cc80, {0x12c86e0, 0xc0000f5350})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.33.0/helper/schema/resource.go:909 +0xa89
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000009b00, {0x1510c08?, 0xc000580c00?}, 0xc0004ae0a0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.33.0/helper/schema/grpc_provider.go:1078 +0xd5c
github.com/hashicorp/terraform-plugin-mux/tf5to6server.v5tov6Server.ApplyResourceChange({{0x1518e60?, 0xc000009b00?}}, {0x1510c08, 0xc000580c00}, 0x0?)
        github.com/hashicorp/terraform-plugin-mux@v0.15.0/tf5to6server/tf5to6server.go:47 +0x54
github.com/hashicorp/terraform-plugin-mux/tf6muxserver.(*muxServer).ApplyResourceChange(0xc0002e10a0, {0x1510c08?, 0xc000580930?}, 0xc0004ae050)
        github.com/hashicorp/terraform-plugin-mux@v0.15.0/tf6muxserver/mux_server_ApplyResourceChange.go:36 +0x193
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ApplyResourceChange(0xc000249ae0, {0x1510c08?, 0xc000580270?}, 0xc0006b0000)
        github.com/hashicorp/terraform-plugin-go@v0.22.1/tfprotov6/tf6server/server.go:846 +0x3d0
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ApplyResourceChange_Handler({0x134eea0, 0xc000249ae0}, {0x1510c08, 0xc000580270}, 0xc00017c800, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.22.1/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:518 +0x1a6
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000179200, {0x1510c08, 0xc0005801e0}, {0x15173b0, 0xc000458000}, 0xc00069a000, 0xc0003a66f0, 0x1b73278, 0x0)
        google.golang.org/grpc@v1.62.1/server.go:1386 +0xdf8
google.golang.org/grpc.(*Server).handleStream(0xc000179200, {0x15173b0, 0xc000458000}, 0xc00069a000)
        google.golang.org/grpc@v1.62.1/server.go:1797 +0xe87
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.62.1/server.go:1027 +0x8b
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 36
        google.golang.org/grpc@v1.62.1/server.go:1038 +0x125

Error: The terraform-provider-proxmox_v0.50.0.exe plugin crashed!

```

**To Reproduce**
Steps to reproduce the behavior:

1. On your root main.tf, declare a child module to your VM creation code, like that  : 

```
module "init_template" {
  source = "./provider/init_template"
}
```

2.  Create a folder (and in my case a sub-folder) to your module and create a variable.tf and put the variables and the main.tf file with the code to create a VM :

```
terraform {
  required_providers {
    proxmox = {
      source = "bpg/proxmox"
      version = "0.50.0"
    }
  }
}

provider "proxmox" {
  endpoint = var.PROXMOX_VE_ENDPOINT
  username = var.PROXMOX_VE_USERNAME
  password = var.PROXMOX_VE_PASSWORD
  insecure = true

  ssh {
    agent = false
  }
}
resource "proxmox_virtual_environment_vm" "template_vm" {
  name        = "template"
  node_name = var.PROXMOX_VE_DEFAULT_NODE
  vm_id     = 9192
  on_boot = false
  started = false

  agent {
    enabled = false
  }

  cpu {
    sockets = 1
    cores = 2
    type = "x86-64-v3"
  }
  memory {
    dedicated = 2048
  }

  initialization {
    ip_config {
      ipv4 {
        address = "${var.TEMPLATE_IP}${var.TEMPLATE_MASK}"
        gateway = var.TEMPLATE_GW
      }
    }

    user_account {
      keys     = [trimspace(var.TEMPLATE_SSH)]
      password = "password"
      username = "ubuntu"
    }
    datastore_id = var.PROXMOX_VE_DEFAULT_DATASTORE 
  }

  network_device {
    bridge = "vmbr0"
  }

  operating_system {
    type = "l26"
  }

}
```

3. Run "terraform init", that should be ok
4. Run "terraform apply" and then "yes" and then issue should occur.


Please also provide a minimal Terraform configuration that reproduces the issue.

```hcl

Terraform v1.7.5
on windows_amd64
+ provider registry.terraform.io/bpg/proxmox v0.50.0
+ provider registry.terraform.io/hashicorp/null v3.2.2
+ provider registry.terraform.io/hashicorp/random v3.6.0
+ provider registry.terraform.io/hashicorp/tls v4.0.5
```

**Expected behavior**
Vm should be up but I don't know why it does not.

**Additional context**

- Clustered Proxmox

--- a/proxmoxtf/resource/vm/vm.go
+++ b/proxmoxtf/resource/vm/vm.go
@@ -2890,7 +2890,9 @@ func vmGetCloudInitConfig(d *schema.ResourceData) *vms.CustomCloudInitConfig {
 			sshKeys := make(vms.CustomCloudInitSSHKeys, len(keys))
 
 			for i, k := range keys {
-				sshKeys[i] = k.(string)
+				if k != nil {
+					sshKeys[i] = k.(string)
+				}
 			}
 
 			initializationConfig.SSHKeys = &sshKeys

